system/jlib/jlog.cpp

```diff
@@ -187,6 +187,60 @@ unsigned getPositionOfField(unsigned logfields, unsigned positionoffield)
 
 // LogMsg
 
+
+static LogMsgJobInfo globalDefaultJobInfo(UnknownJob, UnknownUser);
+static thread_local LogMsgJobInfo defaultJobInfo = globalDefaultJobInfo;
+
+const LogMsgJobInfo unknownJob(UnknownJob, UnknownUser);
+
+void resetThreadLogging()
+{
+    // Note - as implemented the thread default job info is determined by what the global one was when the thread was created.
+    // There is an alternative interpretation, that an unset thread-local one should default to whatever the global one is at the time the thread one is used.
+    // In practice I doubt there's a lot of difference as global one is likely to be set once at program startup
+    defaultJobInfo = globalDefaultJobInfo;
+}
+
+const LogMsgJobInfo & checkDefaultJobInfo(const LogMsgJobInfo & _jobInfo)
+{
+    if (&_jobInfo == &unknownJob)
```

Comment:
I think, better test is if _jobInfo.queryJobID() == UnknownJob

Reply:
Why?

Reply:
Because it will work on a copy of unknownJob (e.g., uninitialised global variables)

Reply:
It will also give wrong results if you tried to set just the user ID (though I don't know what the "right" result would be, or if userId is ever in use or likely to be (should we do similar things with userId as I just did with JobID ?

Reply:
I still don't like the idea of comparing an address of an object, but it's probably ok.